packages/rs-drive-abci/src/execution/validation/state_transition/check_tx_verification/v0/mod.rs

```diff
@@ -0,0 +1,188 @@
+use crate::error::Error;
+use crate::execution::types::execution_event::ExecutionEvent;
+use crate::execution::validation::state_transition::transformer::StateTransitionActionTransformerV0;
+use crate::platform_types::platform::{PlatformRef, PlatformStateRef};
+use crate::platform_types::platform_state::v0::PlatformStateV0Methods;
+use crate::rpc::core::CoreRPCLike;
+use dpp::identity::PartialIdentity;
+use dpp::prelude::ConsensusValidationResult;
+
+use dpp::serialization::Signable;
+use dpp::state_transition::{StateTransition};
+use drive::state_transition_action::StateTransitionAction;
+use dpp::validation::SimpleConsensusValidationResult;
+use dpp::version::{DefaultForPlatformVersion, PlatformVersion};
+use drive::drive::Drive;
+use drive::grovedb::TransactionArg;
+use crate::error::execution::ExecutionError;
+use crate::execution::check_tx::CheckTxLevel;
+use crate::execution::types::state_transition_execution_context::StateTransitionExecutionContext;
+use crate::execution::validation::state_transition::common::asset_lock::proof::AssetLockProofStateValidation;
+use crate::execution::validation::state_transition::common::validate_state_transition_identity_signed::{ValidateStateTransitionIdentitySignature};
+use crate::execution::validation::state_transition::processor::process_state_transition;
+use crate::execution::validation::state_transition::processor::v0::{StateTransitionSignatureValidationV0, StateTransitionStructureValidationV0};
+use crate::execution::validation::state_transition::state_transitions::identity_update::identity_and_signatures::v0::IdentityUpdateStateTransitionIdentityAndSignaturesValidationV0;
+use crate::execution::validation::state_transition::state_transitions::identity_create::identity_and_signatures::v0::IdentityCreateStateTransitionIdentityAndSignaturesValidationV0;
+use crate::execution::validation::state_transition::state_transitions::identity_top_up::identity_retrieval::v0::IdentityTopUpStateTransitionIdentityRetrievalV0;
+pub(super) fn check_tx_state_transition_to_execution_event_v0<'a, C: CoreRPCLike>(
+    platform: &'a PlatformRef<C>,
+    state_transition: StateTransition,
+    check_tx_level: CheckTxLevel,
+    platform_version: &PlatformVersion,
+) -> Result<ConsensusValidationResult<Option<ExecutionEvent<'a>>>, Error> {
+    match check_tx_level {
+        CheckTxLevel::FirstTimeCheck => {
+            if state_transition.requires_check_tx_full_validation() {
+                return Ok(process_state_transition(platform, state_transition, None)?.map(Some));
+            } else {
+                // we need to validate the structure, the fees, and the signature
+                let mut state_transition_execution_context =
+                    StateTransitionExecutionContext::default_for_platform_version(
+                        platform_version,
+                    )?;
+
+                let action = if state_transition.requires_state_to_validate_structure() {
```

Comment:
Will it fetch a contract and validate a document structure as well?

Reply:
Yes, it needs the contract to know what it will need to insert as secondary indexes, as well as how to deserialize. I don't believe it will validate all the document structure rules, for example it shouldn't validate the json schema.

Reply:
Ok, I see